sdk/core/azure-core/DEPENDENCIES.md

```diff
@@ -0,0 +1,419 @@
+# Dependencies
```

Comment:
If we are adding this, we should consider including this in all client libraries. Users generally don't take a direct dependency on core. So, they usually start with a client library and then work their way down to azure-core.